test/Datadog.Trace.Tests/Configuration/ConfigurationSourceTests.cs

```diff
@@ -63,10 +63,14 @@ public static IEnumerable<object[]> GetTestData()
 
             yield return new object[] { ConfigurationKeys.Environment, "staging", CreateFunc(s => s.Environment), "staging" };
 
+            yield return new object[] { ConfigurationKeys.Version, "1.0.0", CreateFunc(s => s.Version), "1.0.0" };
+
+            yield return new object[] { ConfigurationKeys.Service, "web-service", CreateFunc(s => s.ServiceName), "web-service" };
             yield return new object[] { ConfigurationKeys.ServiceName, "web-service", CreateFunc(s => s.ServiceName), "web-service" };
 
             yield return new object[] { ConfigurationKeys.DisabledIntegrations, "integration1;integration2", CreateFunc(s => s.DisabledIntegrationNames.Count), 2 };
 
+            yield return new object[] { ConfigurationKeys.Tags, "k1:v1, k2:v2", CreateFunc(s => s.GlobalTags.Count), 2 };
```

Comment:
I would make sure to have tests that cover:

`k1:`

`:,:,:,:`

`k1:,k2:v2`

etc

Reply:
empty keys are not allowed and should be ignored.

same for empty values, it isn't likely that anyone would want an empty string as a value, so we can ignore if it doesn't have a value set.

Reply:
Good call. This actually exposed an issue where we didn't discard those values. This is fixed in 666f44c8bf0afb24c506e2758cfd440f59e7d2c7